jre_emul/android/platform/libcore/ojluni/src/main/java/sun/reflect/Reflection.java

```diff
@@ -29,14 +29,21 @@
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
+//import dalvik.system.VMStack;
 
 /** Common utility routines used by both java.lang and
     java.lang.reflect */
 
 public class Reflection {
 
-    public static void ensureMemberAccess(Class currentClass,
-                                          Class memberClass,
+   /*  public static Class<?> getCallerClass() {
+        // This method (getCallerClass()) constitutes another stack frame,
+        // so we need to call getStackClass2() rather than getStackClass1().
+        return VMStack.getStackClass2();
+    } */
```

Comment:
Mary's PR (https://github.com/google/j2objc/pull/1380/files) updates two java.util.concurrent.atomic classes to call Reflection.getCallerClass(), which is why I sent the note about you two needing to coordinate with each other. Perhaps you could pair-program an implementation for this method? If I understand its requirements, this can be implemented with the following pseudo-code (not compiled or tested, so your solution may be different):
```
try {
    StackTraceElement[] stack = new Throwable().getStackTrace();
    return Class.forName(stack[2].getClassName());  // index may be wrong, verify with test
} catch (ClassNotFoundException e) {
    // shouldn't happen if the class name is right, but StackTraceElement.getClassName() might return a camel-cased name, null, or ???
} catch (???) {
}
```